frontend/src/components/resourceMap/KubeObjectGlance/NodeGlance.tsx

```diff
@@ -0,0 +1,40 @@
+/*
+ * Copyright 2025 The Kubernetes Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { useSelector } from 'react-redux';
+import { KubeObject } from '../../../lib/k8s/cluster';
+import { RootState } from '../../../redux/stores/store';
+import { KubeObjectGlance } from './KubeObjectGlance';
+
+/**
+ * Generic glance component for rendering node previews in Headlamp's graph view.
+ * Renders KubeObjectGlance for Kubernetes objects or a plugin-provided glance for other nodes.
+ */
+export const NodeGlance = ({ resource }: { resource: any }) => {
```

Comment:
instead of resources, it should accept a `node` property. And it should have a `GraphNode` type instead of `any`